src/main/java/bowling/domain/frame/Frame.java

```diff
@@ -30,14 +36,63 @@ public Frame next() {
     }
 
     public void roll(int numberOfDownPin) {
-        this.pins.down(numberOfDownPin);
+        this.frameDownPins.down(numberOfDownPin);
     }
 
     public boolean hasTurn() {
-        return this.pins.hasTurn();
+        return this.frameDownPins.hasTurn();
     }
 
     public FrameResult getFrameResult() {
-        return new FrameResult(this.pins.getDownPins(), this.pins.getScoreType());
+        return new FrameResult(this.frameDownPins.getDownPins(), this.frameDownPins.getScoreType(), frameScore());
+    }
+
+    public Score frameScore() {
+        if (!isFinished()) {
+            return Score.create(0, ScoreType.READY);
+        }
+
+        if (isLast() && isFinished()) {
+            return Score.create(this.frameDownPins.sum(), ScoreType.NORMAL);
+        }
+
+        int nextBowlCount = this.frameDownPins.getScoreType().getBonusBowlCount();
+        List<Integer> nextDownPins = getNextDownPins(nextBowlCount);
+        if (nextDownPins.size() < nextBowlCount) {
+            return Score.create(0, ScoreType.READY);
+        }
+
+        int score = this.frameDownPins.sum() + nextDownPins.stream().reduce(0, Integer::sum);
```

Comment:
```suggestion
        int score = this.frameDownPins.sum() + nextDownPins.stream()
                                                                                                    .reduce(0, Integer::sum);
```
개인차가 있겠지만 스트림은 기능단위로 줄바꿈하여 사용하는것을 권장드립니다.